design/hardware-classification-controller/support-for-new-parameters-hwcc.md

```diff
@@ -0,0 +1,243 @@
+         * [Risks and Mitigation](#risks-and-mitigations)
+      * [Design Details](#design-details)
+         * [Work Items](#work-items)
+         * [Dependencies](#dependencies)
+         * [Test Plan](#test-plan)
+      * [References](#references)
+
+## Summary
+ 
+Implement support for more parameters in HWCC which will help to 
+classify a baremetal host on more possible configurations. 
+ 
+## Motivation
+
+We have hardware classification controller which validates and 
+classifies the hosts based on user hardware requirements.
+We would like to add more parameters which helps user to 
+identify the ideal host. This feature helps user to avoid runtime failures
+and also increases the performance for workload deployments.
+
+### Goals
+
+Automatically classify bare metal hosts based on user-provided rules
+in a way that other controllers can use, without having to evaluate
+all of those rules themselves.
+
+We will carry testing of HWCC only on Dell hardware. Testing of HWCC on 
+hardware from other vendors will not be goal of this proposal.
+
+## Proposal
+ 
+We are submitting proposal based on the issue raised in HWCC: 
+[Issue 53](https://github.com/metal3-io/hardware-classification-controller/issues/53)
+
+### Investigation Details
+
+Add support for new parameters:
+
+1. CPU
+
+    `arch` will come under CPU parameter.
+
+    User can provide `arch` type based on requirement.
+
+    Investigation Details: Below is sample introspection data for CPU.
+     `arch` parameter will be added under `cpu` section.
+    Types of architecture are 32-bit (x86) and 64-bit 
+    (x86-64, IA64, and AMD64).
+
+    ```yaml
+    hardware:
+        cpu:
+           arch: "x86_64"
+    ```
+
+2. Disk
+
+    `type` will come under disk parameter.
+
+    User can provide multiple disk type (RAID/HDD/SSD/NVMe), count and
+     individual size based on requirement.
+
+    Investigation Details: Below is sample introspection data for Disk.
+
+     1. HDD - Identified using `rotational` parameter's value as true
+      and `hctl` value as "0:0:N:0".  
+     2. SSD - Identified using `rotational` parameter's value as false
+      and `hctl` value as "0:0:N:0" or "N:0:0:0".
+     3. NVMe - Identified with the combination of `model` name and
+      `rotational` parameter's value as false.
+     4. RAID - Identified with the combination of
+
+        a) `rotational` parameter's value as true.
+
+        b) If `hctl` have value as "N:0:0:0" or "0:N:0:0".
+
+     ```yaml
+     hardware:
+        disks:
+          - name: "/dev/sda"
+            model: "KPM5XVUG3T84"
+            rotational: "true"
+            hctl: "0:0:N:0"
+     ```
+3. NIC
+
+     Adding new parameter `vendorName` under NIC. User can provide NIC
+      requirement in the form of `vendor`.
+
+     Investigation details: Below is sample introspection data for NICs.
+      `model` field under `nics` contains combination of Vendor ID and
+       Product ID.
```

Comment:
Both of these are strings. Do we need to support sub-string matching or regular expressions?

Reply:
Yes, we are using sub-string matching for Vendor name.

Reply:
It's anyway hardcoded values, Its fine to keep it as a string.

Reply:
Either is fine with me. We should document our decision here.